### PR TITLE
Only follow full DFG flow in `followNextDFGEdgesUntilHit` and similar

### PR DESCRIPTION
**Pull request by oxisto**
We had a problem that the extension functions, such as `followNextDFGEdgesUntilHit` were following all DFG flows, including partial ones. This lead to unexpected results.

### CODE REVIEW

File: cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt [hunk 1]
```diff
@@ -233,13 +233,13 @@ fun Node.followPrevDFGEdgesUntilHit(predicate: (Node) -> Boolean): FulfilledAndF
 
     while (worklist.isNotEmpty()) {
         val currentPath = worklist.removeFirst()
-        if (currentPath.last().prevDFG.isEmpty()) {
+        if (currentPath.last().prevFullDFG.isEmpty()) {
```
Comment (konradweiss) [inline cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt:236]: I think we either need to rename the function reflecting that we are only following full dfgs or make it configurable. if the configuration has the default value full, we may have the same problem with confusing naming, so the default should not restrict the granularity.

**Review by konradweiss — CHANGES_REQUESTED** [review]

File: cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt (hunk 1 above)
Comment (oxisto) [reply cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt:236]: Yes that is true, ok that sounds like a potentially bigger issue/discussion. Can we put this on the agenda for the dev meeting on 4/4/24?.

Comment (konradweiss) [reply cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt:236]: I thought about simply renaming the function to `followPrevFullDFGEdgesUntilHit` as a minimum requirement to merge this PR as it sound rather urgent and then merge it. If there are more issues or better solutions we can put it on the agenda.

Comment (oxisto) [reply cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt:236]: Sounds also reasonable to me

**Review by KuechA — APPROVED** [review]

**Review by konradweiss — APPROVED** [review]

**Pull request merged**